week02/Group Anagrams.py

```diff
@@ -0,0 +1,21 @@
+#O(n),O(1)
+class Solution(object):
+    def groupAnagrams(self, strs):
+        """
+        :type strs: List[str]
+        :rtype: List[List[str]]
+        """
+        charl = {'a':2,'b':3,'c':5,'d':7,'e':11,'f':13,'g':17,'h':19,'i':23,
+             'j':29,'k':31,'l':37,'m':41,'n':43,'o':47,'p':53,'q':59,'r':61,
+             's':67,'t':71,'u':73,'v':79,'w':83,'x':89,'y':97,'z':101}
+        hashAllChars = {}
+        for i in strs:
+            r = 1
+            for j in i:
+                r *= charl[j]
+            if r in hashAllChars:
+                hashAllChars[r].append(i)
+            else:
+                hashAllChars[r] = [i]
```

Comment:
- Worst possible variable naming. I had to literally spend 5 minutes to identify what `i` represents. Improve your variable naming.
- It does not look like `O(1)` in memory.